[Feat] expand docs: explain 1000-rec order limit (#161)
libremfg/rhize-docs: content/how-to/gql/filter.md

{{% /tab %}}
{{< /tabs >}}

#### Avoid using @cascade with the [`order`]({{< relref "query#order" >}}) argument

The `order` argument returns only the first 1000 records of the query.
If a record matches the `@cascade` filter but comes after these first 1000 records, the API does not return it.

For example, this query logic works as follows:
1. Return the first 1000 records of equipment as ordered by `effectiveStart`.

> 1. From these one 1000 records, return only the equipment items that are part of `parentEquipment1`.

Author: ```suggestion
1. From these 1000 records, return only the equipment items that are part of `parentEquipment1`.
```

> Such

Author: ```suggestion
```

```graphql
query($filter: EquipmentFilter){
  queryEquipment (filter: { order: {desc:effectiveStart}) @cascade{
    id
    isPartOf (filter: {id:{eq:"parentEquipment1"}}) {
      id
    }
  }
}
```

> This omission can be surprising, so avoid `@cascade` with the `order` argument.

Author: ```suggestion
This behavior can be surprising and undesirable, so avoid `@cascade` with the `order` argument.
```
